### PR TITLE
Prepare for upcoming deprecation of Ethereum testnets

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves https://github.com/brave/brave-browser/issues/25654

**Note:** This should only be merged after devops has configured the Sepolia proxy.
## Changes
* Remove references to Rinkeby, Ropsten, and Kovan testnets as they will be deprecated soon.

* Add Sepolia Ethereum testnet as a preloaded network

* Migrate test network used by Swap Service to Goerli

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* On all platforms, have a deprecated test network selected, upgrade the browser to include this change, verify that Ethereum mainnet is now selected and that kBraveWalletUserAssetsAddPreloadingNetworksMigrated is now set to true.  Ropsten, Kovan, and Rinkeby networks should no longer be a preloaded network selectable as a user
* Verify that Sepolia is now a preloaded network, and can be used (e.g. to fetch balances)

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_constants.h [hunk 1]
```diff
@@ -940,9 +938,9 @@ constexpr webui::LocalizedString kLocalizedStrings[] = {
 };
 
 // 0x swap constants
-constexpr char kRopstenSwapBaseAPIURL[] = "https://ropsten.api.0x.org/";
+constexpr char kGoerliSwapBaseAPIURL[] = "https://goerli.api.0x.org/";
```
Comment (nvonpentz) [inline components/brave_wallet/browser/brave_wallet_constants.h:941]: Confirmed 0x supports Goerli https://docs.0x.org/0x-api-swap/api-references

File: browser/brave_wallet/brave_wallet_service_unittest.cc [hunk 2]
```diff
@@ -72,7 +72,7 @@ const char token_list_json[] = R"(
    }
   })";
 
-const char ropsten_list_json[] = R"(
+const char goerli_list_json[] = R"(
```
Comment (nvonpentz) [inline browser/brave_wallet/brave_wallet_service_unittest.cc:75]: For context, this was previously named incorrectly  - the chain ID 0x2a refers to the Kovan network.

**Review by kdenhartog — APPROVED** [review]
Security wise LGTM

File: components/brave_wallet_ui/constants/types.ts [hunk 3]
```diff
@@ -734,10 +731,7 @@ export const SupportedOnRampNetworks = [
 ]
 
 export const SupportedTestNetworks = [
-  BraveWallet.RINKEBY_CHAIN_ID,
```
Comment (onyb) [inline components/brave_wallet_ui/constants/types.ts:737]: Missing entry for Sepolia?

**Review by onyb — APPROVED** [review]
Left a minor comment, but LGTM otherwise.

File: components/brave_wallet_ui/constants/types.ts (hunk 3 above)
Comment (nvonpentz) [reply components/brave_wallet_ui/constants/types.ts:743]: Added thanks

File: components/brave_wallet/browser/json_rpc_service.cc [hunk 4]
```diff
@@ -228,6 +228,25 @@ void JsonRpcService::MigrateMultichainNetworks(PrefService* prefs) {
...
+// static
+void JsonRpcService::MigrateDeprecatedEthereumTestnets(PrefService* prefs) {
+  if (prefs->GetBoolean(kBraveWalletDeprecateEthereumTestNetworksMigrated))
+    return;
+
+  if (prefs->HasPrefPath(kBraveWalletSelectedNetworks)) {
+    DictionaryPrefUpdate update(prefs, kBraveWalletSelectedNetworks);
+    auto& selected_networks_pref = update.Get()->GetDict();
+    const std::string* selected_eth_network =
+        selected_networks_pref.FindString(kEthereumPrefKey);
```
Comment (yrliou) [inline components/brave_wallet/browser/json_rpc_service.cc:240]: if (!selected_eth_network)
  return;

Comment (nvonpentz) [reply components/brave_wallet/browser/json_rpc_service.cc:240]: Added, thanks

**Comment by srirambv** [comment]
Verification passed on <html><head></head>
<body>


Brave | 1.46.35 Chromium: 106.0.5249.91 (Official Build) nightly (64-bit)
-- | --
Revision | `fa96d5f07b1177d1bf5009f647a5b8c629762157-refs/branch-heads/5249@{#707}`
OS | Windows 11 Version 22H2 (Build 22623.730)


</body>
</html>

- Verified steps from PR 
- Verified deprecated networks are removed after update
- Verified Sepolia & Goreli are the only two ETH test networks available after upgrade
- Verified Sepolia is added as default for new wallets and fetches balance

Upgrade | Sepolia Balance
-- | --
<video src=https://user-images.githubusercontent.com/17010094/194215899-ef34660e-7d67-4fbe-a345-0658825b23d9.mov> | image

**Comment by rishiraj88** [comment]
Thanks

File: browser/brave_wallet/brave_wallet_service_unittest.cc (hunk 2 above)
Comment (ho-lapang) [reply browser/brave_wallet/brave_wallet_service_unittest.cc:75]: 182570408a1f25ab2731ef5f283b918df9b9f956-refs/branch-heads/5249_91@{#6

Comment (ho-lapang) [reply browser/brave_wallet/brave_wallet_service_unittest.cc:75]: What is that

**Pull request merged**